cnd/src/swap_protocols/halight.rs

```diff
@@ -243,7 +263,23 @@ where
     })
 }
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Copy, Clone, Debug)]
 pub struct Params {
-    pub secret_hash: SecretHash,
+    pub ethereum_identity: EthereumIdentity,
+    pub ethereum_absolute_expiry: Timestamp,
+    pub lightning_identity: identity::Lightning,
+    pub lightning_cltv_expiry: Timestamp,
+    pub lightning_amount: asset::Bitcoin,
+}
+
+impl From<HanEtherereumHalightBitcoinCreateSwapParams> for Params {
```

Comment:
:mega: If you want to do an impl like this, it shouldn't live in this module because you are creating a circular dependency here: The code that uses defines `HanEtherereumHalightBitcoinCreateSwapParams` also uses `Params` so both modules import from each other. Also `HanEtherereumHalightBitcoinCreateSwapParams` is not halight-specific but knows "more" about the codebase than this particular module.

Reply:
I had a look and only facade is importing halight. I cant see any circular dependency?

Reply:
Okay, forget the circular dependency talk :D
This `impl` deals with more than just halight (the current module) and should be defined in a different place IMO :)
Probably wherever `HanEtherereumHalightBitcoinCreateSwapParams` is defined makes the most sense!

You can see it as: conceptually, the halight module doesn't (and shouldn't) "know" in which swaps it is going to be involved. This is a concern of code that composes those modules together.